pkg/services/alerting/engine.go

```diff
@@ -204,6 +205,9 @@ func (e *AlertingService) processJob(attemptID int, attemptChan chan int, cancel
 			}
 		}
 
+		resultHandleCtx, resultHandleCancelFn := context.WithTimeout(context.Background(), resultHandleTimeout)
```

Comment:
This line should not use `context.Background()`. it should use `alertCtx`

Reply:
As https://golang.org/src/context/context.go,  if current deadline is already sooner than the new one then it will return a cancel context and deadline is not updated.
(it will be meet deadline soon.)

My intention is to have the result handler have enough deadlines to perform its job.

Below is I tested.
```
Current Time : 2019-01-15 19:17:24.456081 +0900 KST m=+2.005557500
before derive alertctx deadline  : 2019-01-15 19:17:25.451779 +0900 KST m=+3.001252721

after derive background case deadline : 2019-01-15 19:17:27.456328 +0900 KST m=+5.005804517 
after derive alertctx case deadline : 2019-01-15 19:17:25.451779 +0900 KST m=+3.001252721 
```

Reply:
I'm sorry I got it all wrong. I thought `alertCtx` was the alert engines context. Sorry about the confusion!

Reply:
No problem! Thanks for reviewing. 👍

Reply:
I think it should be possible to add some tests in https://github.com/grafana/grafana/blob/master/pkg/services/alerting/engine_test.go

`alertTimeout` could be overwritten to something very short. and the test could block a little bit longer.

if the test endups taking to long we could add the `// +build integration` build flag in the top of the file to only run it in CI or when developers want to run all tests.

Reply:
You are right. 
I was writing some test code in my repository. https://github.com/sharkpc0813/grafana/commit/f74e10beeb43cd4cf65ef6fc445cbb6749605863
(I wonder that my test codes are proper for this project.)
Let me push that when I'm ready. This will not take much time.